src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-config-modal/rgw-config-modal.component.ts

```diff
@@ -0,0 +1,101 @@
+
+@Component({
+  selector: 'cd-rgw-config-modal',
+  templateUrl: './rgw-config-modal.component.html',
+  styleUrls: ['./rgw-config-modal.component.scss'],
+  providers: [RgwBucketEncryptionModel]
+})
+export class RgwConfigModalComponent implements OnInit{
+  readonly vault_address = /^((https?:\/\/)|(www.))(?:([a-zA-Z]+)|(\d+\.\d+.\d+.\d+)):\d{4}$/
+
+
+  kms_providers: string[];
+
+  configForm: CdFormGroup;
+
+  @Output()
+  submitAction = new EventEmitter();
+  auth_methods: string[];
+  secret_engines: string[];
+
+  constructor(
+    private formBuilder: CdFormBuilder,
+    public activeModal: NgbActiveModal,
+    public actionLabels: ActionLabelsI18n,
+    private rgwBucketService: RgwBucketService,
+    private rgwEncryptionModal: RgwBucketEncryptionModel,
+    private notificationService: NotificationService,
+  ) {
+    this.createForm();
+   }
+  ngOnInit(): void {
+    this.kms_providers = this.rgwEncryptionModal.kms_providers;
+    this.auth_methods = this.rgwEncryptionModal.auth_methods;
+    this.secret_engines = this.rgwEncryptionModal.secret_engines;
+  }
+
+  createForm() {
+    this.configForm = this.formBuilder.group({
+      address: [null, [Validators.required, CdValidators.custom('vaultPattern', (value: string) => {
+        if (_.isEmpty(value)) {
+          return false;
+        }
+        return !this.vault_address.test(value);
+      })]],
+      kms_provider: ['vault', Validators.required],
+      encryptionType: [null, Validators.required],
+      auth_method: [null, Validators.required],
+      secret_engine: [null, Validators.required],
+      secret_path: ['/'],
+      namespace: [null],
+      token: [null, [CdValidators.requiredIf({
+        auth_method: 'token',
+      })]],
+      ssl_cert: [null, CdValidators.sslCert()],
+      client_cert: [null, CdValidators.pemCert()],
+      client_key: [null, CdValidators.sslPrivKey()],
+      role: [null, [CdValidators.requiredIf({
+        auth_method: 'agent',
+      })]],
+      kmsEnabled: [{ value: false }],
+      s3Enabled: [{ value: false }],
+    });
+  }
+
+  fileUpload(files: FileList, controlName: string) {
+    const file: File = files[0];
+    const reader = new FileReader();
+    reader.addEventListener('load', () => {
+      const control: AbstractControl = this.configForm.get(controlName);
+      control.setValue(file);
+      control.markAsDirty();
+      control.markAsTouched();
+      control.updateValueAndValidity();
+    });
+  }
+
+  onSubmit() {
+    const values = this.configForm.value;
+    console.log(values);
+      this.rgwBucketService.setEncryptionConfig(values['encryptionType'], values['kms_provider'], values['auth_method']
+      , values['secret_engine'], values['secret_path'], values['namespace'], values['address'], values['token'], values['owner'], values['ssl_cert'], values['client_cert'], values['client_key']).subscribe(() => {
+        this.notificationService.show(
```

Comment:
need to handle the error too.. when an error happens set the form with errors and not close it probably?